concrete_biopython/BioCircuit.py

```diff
@@ -107,11 +101,11 @@ def __init__(self,
             # set the alphabet if any to be able to call SeqWrapper.maxInteger()
             SeqWrapper.setAlphabet(self._alphabet)
 
-        # Create the circuit integer seq_list from the seq
+        # Create the circuit inputset from the seq lengths and the maximum integer
         # Use SeqWrapper.maxInteger() to know the maximum integer that can be
         # used to represent a character in FheSeq obects        
-        inputset = [tuple([np.random.randint(0, SeqWrapper.maxInteger()+1, size=(len(seq),))
-                            for seq in seq_list]) for _ in range (300)]
+        inputset = [tuple([np.random.randint(0, SeqWrapper.maxInteger()+1, size=(len_seq,))
+                            for len_seq in len_seqs]) for _ in range (300)]
```

Comment:
But I still think you could accept a `inputset: Option[List[Tuple[Union[Seq, MutableSeq]]]] = None` and convert it to an integer inputset if it exist.